### PR TITLE
Login server backend

### PR DESCRIPTION
**Pull request by m4ndycheung**

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -33,3 +34,24 @@ app.post("/pupilRecord", (req, res) => {
...
+
+// endpoint to validate user credentials
+app.post("/login", (req, res) => {
+  console.log(req.body);
+
+  const tempUsername = req.body.teacherUsername;
+  const tempPassword = req.body.teacherPassword;
+
+  // check if matches specific string
+  if (tempUsername === "benning123") {
```
Comment (alinosratipour) [inline server/server.js:52]: don't we want to check for  a password

File: server/server.js [hunk 2]
```diff
@@ -33,3 +34,24 @@ app.post("/pupilRecord", (req, res) => {
...
+app.get("/login", (req, res) => {
+  console.log("Hey there");
+  res.status(200).send("info sent to user");
+});
+
+// endpoint to validate user credentials
+app.post("/login", (req, res) => {
+  console.log(req.body);
+
+  const tempUsername = req.body.teacherUsername;
```
Comment (alinosratipour) [inline server/server.js:48]: we can avoid repeat code like this
 const { teacherUsername, teacherPassword } = req.body;

File: server/server.js [hunk 3]
```diff
@@ -33,3 +34,24 @@ app.post("/pupilRecord", (req, res) => {
     pupilName: reqPupilName,
   });
 });
+
+// endpoint to display login page
+app.get("/login", (req, res) => {
```
Comment (alinosratipour) [inline server/server.js:39]: we can have function for login then call it like this 
  ```
const validateUser= (req, res) => {
  const { teacherUsername, teacherPassword } = req.body;

  // Check if the username  and password matches a specific string
  if (teacherUsername === "benning123" && teacherPassword === "yourPasswordHere") {
    res.status(200).json({ message: "login details correct", teacherID: "3" });
  } else {
    res.status(401).json({ error: "incorrect login details" });
  }
};

app.post("/login", validateUser);
```

File: server/server.js (hunk 1 above)
Comment (m4ndycheung) [reply server/server.js]: Yes we do, I forgot to add this. Thanks, Ali - added now :)

File: server/server.js (hunk 2 above)
Comment (m4ndycheung) [reply server/server.js]: Updated in the code, thanks!

File: server/server.js (hunk 3 above)
Comment (m4ndycheung) [reply server/server.js]: Had a meeting with team, we agree, we'll take this approach going forward. Thanks for the review!

**Pull request merged**